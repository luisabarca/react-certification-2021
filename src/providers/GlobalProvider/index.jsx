import React, { useContext, useState, useCallback } from 'react';
import { createTheme, CssBaseline, ThemeProvider } from '@material-ui/core';

export const GlobalContext = React.createContext(null);

export const useGlobalContext = () => {
  const context = useContext(GlobalContext);

  if (!context) {
    throw new Error('GlobalContext not available');
  }

  return context;
};

export const GlobalProvider = ({ children }) => {
  const [query, setQuery] = useState('Wizeline');
  const [theme, setTheme] = useState('dark');

  const toggleTheme = useCallback(() => {
    setTheme(theme == 'light' ? 'dark' : 'light');
  }, [theme]);

  const selectedTheme = createTheme({
    palette: {
      type: theme,
    },
  });

  return (
    <GlobalContext.Provider value={{ query, setQuery, theme, setTheme, toggleTheme }}>
      <ThemeProvider theme={selectedTheme}>
      <CssBaseline />
        {children}
      </ThemeProvider>
    </GlobalContext.Provider>
  );
};

export default GlobalProvider;
