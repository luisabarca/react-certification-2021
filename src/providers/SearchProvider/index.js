import React, { useContext, useState } from 'react';

export const SearchContext = React.createContext(null);

export const useSearch = () => {
  const context = useContext(SearchContext);

  if (!context) {
    throw new Error('SearchProvider not available');
  }

  return context;
};

export const SearchProvider = ({ children }) => {
  const [query, setQuery] = useState('Wizeline');

  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <SearchContext.Provider value={{ query, setQuery }}>
      {children}
    </SearchContext.Provider>
  );
};

export default SearchProvider;
