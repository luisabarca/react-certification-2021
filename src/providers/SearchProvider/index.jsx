import React, { useContext, useState } from 'react';

export const SearchContext = React.createContext(null);

export const useSearch = () => {
  const context = useContext(SearchContext);

  if (!context) {
    throw new Error('SearchProvider not available');
  }

  return context;
};

export const SearchProvider = ({ children }) => {
  const [query, setQuery] = useState('Wizeline');

  return (
    <SearchContext.Provider value={{ query, setQuery }}>
      {children}
    </SearchContext.Provider>
  );
};

export default SearchProvider;
