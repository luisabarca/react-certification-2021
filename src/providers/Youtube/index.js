import React, { useCallback, useContext, useEffect, useState } from 'react';

const YOUTUBE_API_KEY = process.env.REACT_APP_YOUTUBE_API_KEY;
const YoutubeApiContext = React.createContext(null);

export const useYoutubeApi = () => {
  const context = useContext(YoutubeApiContext);

  if (!context) {
    throw new Error('YoutubeApiProvider not available');
  }

  return context;
};

const loadClientScript = () => {
  const script = document.createElement('script');
  script.src = 'https://apis.google.com/js/client.js';

  return script;
};

export const YouTubeApiProvider = ({ children }) => {
  const { gapi } = window;
  const [ready, setReady] = useState(false);

  useEffect(() => {
    const script = loadClientScript();

    script.onload = () => {
      gapi.load('client', () => {
        // Set API KEY.
        gapi.client.setApiKey(YOUTUBE_API_KEY);
        gapi.client.load('youtube', 'v3', () => {
          setReady(true);
        });
      });
    };

    document.body.appendChild(script);
  }, [gapi]);

  const searchVideos = useCallback(async (query) => {
    const options = {
      part: ['snippet'],
      maxResults: 25,
      q: query,
    };

    try {
      const response = await window.gapi.client.youtube.search.list(options);
      return response.result;
    } catch (error) {
      return null;
    }
  }, []);

  const getVideo = useCallback(async (videoId) => {
    const options = {
      part: ['snippet,contentDetails,statistics'],
      id: videoId,
    };

    try {
      const response = await window.gapi.client.youtube.videos.list(options);
      return response.result;
    } catch (error) {
      return null;
    }
  }, []);

  if (!ready) return null;

  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <YoutubeApiContext.Provider value={{ searchVideos, getVideo }}>
      {children}
    </YoutubeApiContext.Provider>
  );
};

export default YouTubeApiProvider;
