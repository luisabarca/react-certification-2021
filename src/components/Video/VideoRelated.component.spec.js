/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

// CUT.
import VideoRelated from './VideoRelated.component';

import { SearchContext } from '../../providers/SearchProvider';

// Returns 3 videos
jest.mock('../../utils/hooks/useVideos');

jest.mock('react-router-dom', () => {
  return {
    Link: jest.fn(({ children }) => <div role="link">{children}</div>),
  };
});

const currentVideo = {
  kind: "youtube#searchResult",
  etag: "G92NWfwsZsCq390K5iD559vVeII",
  id: {
    kind: "youtube#video",
    videoId: "nmXMgqjQzls"
  },
  snippet: {
    publishedAt: "2019-09-30T23:25:15Z",
    channelId: "UCPGzT4wecuWM0BH9mPiulXg",
    title: "Video Tour | Welcome to Wizeline Guadalajara",
    description: "Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office.\n\nIn 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, Mexico, home to over 400 employees. Wizeline has drawn attention for designing an office and culture focused on employee wellness and cultivating talent, earning a Super Espacios award by WeWork, Expansion Mexico, and Top Companies.\n\nRead more about the office here: https://www.wizeline.com/creating-first-class-facilities/\n\nIf you're interested in working with Wizeline and want to experience the culture and tour our offices live, contact us at www.wizeline.com/contact/",
    thumbnails: {
      default: {
        url: "https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg",
        width: 120,
        height: 90
      },
      medium: {
        url: "https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg",
        width: 320,
        height: 180
      },
      high: {
        url: "https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg",
        width: 480,
        height: 360
      },
      standard: {
        url: "https://i.ytimg.com/vi/nmXMgqjQzls/sddefault.jpg",
        width: 640,
        height: 480
      },
      maxres: {
        url: "https://i.ytimg.com/vi/nmXMgqjQzls/maxresdefault.jpg",
        width: 1280,
        height: 720
      }
    },
    channelTitle: "Wizeline",
    liveBroadcastContent: "none",
    publishTime: "2019-09-30T23:25:15Z"
  }
};

const renderVideoRelated = (snap) => {
  const isSnap = snap || false;
  const component = <VideoRelated video={currentVideo} />;

  if (isSnap) {
    return renderer.create(component);
  }

  return render(component);
};


describe('VideoRelated Component', () => {
  beforeEach(() => {
    renderVideoRelated();
  });

  it('should match snapshot', () => {
    const snap = renderVideoRelated(true);

    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('should render a link', () => {
    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
  });

  it('should render a thumbnail image', () => {
    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
  });

  it('should render a title', () => {
    const title = screen.getByRole('heading');
    expect(title).toBeInTheDocument();
  });
});
