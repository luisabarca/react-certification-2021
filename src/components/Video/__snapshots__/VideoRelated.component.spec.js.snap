// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoRelated Component should match snapshot 1`] = `
<div
  role="link"
>
  <div
    className="sc-bdnxRM jDfQMU"
  >
    <div
      className="sc-dlnjwi kFVqTn"
      role="img"
    />
    <h2
      className="sc-gtsrHT djjKNm makeStyles-title-1"
    >
      Video Tour | Welcome to Wizeline Guadalajara
    </h2>
  </div>
</div>
`;
